include/seastar/net/api.hh

```diff
@@ -169,6 +169,14 @@ struct session_dn {
     sstring issuer;
 };
 
+  /// Information about a certificate
+struct cert_info {
+    static constexpr size_t bytes_inline_size = 31;
+    using bytes = basic_sstring<uint8_t, uint32_t, bytes_inline_size, false>;
```

Comment:
Why introduce a special type? Yes it saves a byte, but it forces a new instantiation of basic_sstring for a weak reason.

sstring should be fine.

Reply:
Less to save a memory and more to express that this is an owned buffer of raw bytes as opposed to a legible string. Felt relevant here because one could reasonably expect either representation for a serial number. 

Anyway, if `sstring` is the preferred type for this usage, I'm happy to roll it back, perhaps expanding the doc comment.

Reply:
Another alternative is std::vector\<std::byte>. I recognize the basic_sstring specialization, but I prefer not to introduce ad-hoc types. It would be different if seastar had a bytes type (well it does, temporary_buffer\<char>, but it's not suitable here).

Reply:
>std::vector<std::byte>

Yeah, I like that better. I considered temporary_buffer as well, but as noted not a great fit.